File: VB/WpfApplication1/MyChartControl/SizeHelper.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Linq
Imports System.Windows
Imports System.Windows.Media
Imports System.Collections.Generic

Namespace MyChartControl
	Public Module SizeHelper
		Sub New()
		End Sub
		<System.Runtime.CompilerServices.Extension> _
		Public Function BoundsRelativeTo(ByVal element As FrameworkElement, ByVal relativeTo As Visual) As Rect
			Return element.TransformToVisual(relativeTo).TransformBounds(New Rect(0, 0, element.ActualWidth, element.ActualHeight))
		End Function
	End Module
End Namespace

File: CS/WpfApplication1/MyChartControl/SizeHelper.cs
using System;
using System.Linq;
using System.Windows;
using System.Windows.Media;
using System.Collections.Generic;

namespace MyChartControl
{
    public static class SizeHelper
    {
        public static Rect BoundsRelativeTo(this FrameworkElement element,
                                         Visual relativeTo)
        {
            return
              element.TransformToVisual(relativeTo)
                     .TransformBounds(new Rect(0, 0, element.ActualWidth, element.ActualHeight));
        }
    }
}
